lib/_http_client.js

```diff
@@ -68,7 +68,8 @@ function ClientRequest(options, cb) {
     }
     // Explicitly pass through this statement as agent will not be used
     // when createConnection is provided.
-  } else if (!(agent instanceof Agent.Agent)) {
+  } else if (!(agent instanceof Agent.Agent) &&
+            typeof agent.addRequest !== 'function') {
```

Comment:
Also needs a change to the error message two lines below, which will make this semver-major.